### ISSUE TITLE
Vimtex error using input/include within a subfile

### ISSUE THREAD

**Issue by douglasgscofield** [lervag/vimtex#1151]
Conditions:

* `main.tex`: main file, uses `subfiles` package to load another file with `\subfile{ch1}`.
* `ch1.tex`: `subfiles`-compatible file loaded by `main.tex`.  Loads another file with `\input{ch1_first}`.
* `ch1_first.tex`: standard LaTeX file with no special formatting (e.g., no `subfiles` wrappers) that is loaded into `ch1.tex`

Loading `ch1_first.tex` into Vim with the latest vimtex plugin loaded produces the following error:

~~~~
Error detected while processing function vimtex#init[4]..<SNR>61_init_state[1]..vimtex#state#init[1]..<SNR>72_get_main[39]..<SNR>72_get_main_latexmain[3]..<SNR>72_file_reaches_current:
line    4:
E341: Internal error: lalloc(0, )
~~~~

The error does not appear loading either of the other two files.  Touching a `main.tex.latexmain` file has no effect.

The three files described above:

`main.tex`:
```tex
\documentclass{article}
\usepackage{subfiles}

\begin{document}

\subfile{ch1}

\end{document}
```

`ch1.tex`:
```tex
\documentclass[main.tex]{subfiles}
\begin{document}

\section{Introductory topics}

\input{ch1_first}

\end{document}
```

`ch1_first.tex`:
```tex
\subsection{First section}

This is the first section.
```

Save the above as three separate files.

**Comment by lervag** [comment]
I was not able to reproduce your issue, but I think I understand what was wrong anyway. So I think I've fixed it. Please test.

**Comment by douglasgscofield** [comment]
Yes, this fixes it: no more error produced when loading. I tested with both `\input{ch1_first}` and `\include{ch1_first}`.  Thank you!

**Comment by lervag** [comment]
No problem! And thank you for contributing by raising the issue!
